src/net/sourceforge/kolmafia/textui/Parser.java

```diff
@@ -864,68 +894,49 @@ else if ( scope != null && scope.findVariable( variableName.content ) != null )
 			result = new Variable( variableName.content, t );
 		}
 
-		this.readToken(); // If parsing of Identifier succeeded, go to next token.
-		// If we are parsing a parameter declaration, we are done
-		if ( scope == null )
-		{
-			if ( this.currentToken().equals( "=" ) )
-			{
-				throw this.parseException( "Cannot initialize parameter " + variableName );
-			}
-			return result;
-		}
+		this.readToken(); // read name
 
-		// Otherwise, we must initialize the variable.
+		return result;
+	}
 
-		Value rhs;
+	/**
+	 * Parses the right-hand-side of a variable definition. It is assumed that the caller expects
+	 * an expression to be found, so this method never returns null.
+	 */
+	private Value parseInitialization( final Type t, final BasicScope scope )
```

Comment:
Hm. If you really wanted access to the variable name, I suppose you could pass in the Variable (or VariableReference) here instead of just the Type. Doesn't make sense to initialize something other than a variable, anyhow.

Reply:
I'm mostly reluctant to add a parameter for a single, edge-case purpose.
At the same time, I guess it would still make sense, since that Variable is tied to the assignment...

At at the same the same time time, one of the end-results of the next step is to add the exact location of the errors in the error message, so even without the name of the variable, people will still be able to tell which expression brought the error...

Reply:
Oh, I meant in lieu of the Type, since the Variable(Reference) knows what its type is.

Reply:
Oh, yeah, that works